### PR TITLE
100% coverage on connections

### LINKED ISSUE #272: Reach 100% test coverage on `aea.connections` module
**Is your feature request related to a problem? Please describe.**
Test coverage on aea.connections module is NOT 100%.

**Describe the solution you'd like**
100% coverage on aea.connections

**Describe alternatives you've considered**
N/A

**Additional context**
None

### PR DESCRIPTION
**Pull request by Totoual**
## Fixes

Covers the aea/connections.
#272 
#311 

- [x] aea/connections/local/connection.py
- [x] aea/connections/oef/connection.py
- [x] aea/connections/scaffold/connection.py
- [x] aea/connections/stub/connection.py

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments
None

### CODE REVIEW

File: tests/test_connections/test_local/test_search_services.py [hunk 1]
```diff
@@ -84,6 +86,24 @@ def teardown_class(cls):
...
+        cls.public_key = "mailbox"
+        cls.connection = OEFLocalConnection(cls.public_key_1, cls.node)
+        cls.mailbox = OEFLocalConnection(cls.public_key, cls.node)
+        cls.mailbox.connect()
+        cls.connection.connect()
+        cls.data_model = DataModel("foobar", attributes=[])
+
+    def test_registering_agent(self):
+        service_description = Description({"foo": 1, "bar": "baz"}, data_model=self.data_model)
+        self.connection.channel
```
Comment (DavidMinarsch) [inline tests/test_connections/test_local/test_search_services.py]: This is not testing anything

File: tests/test_connections/test_local/test_search_services.py [hunk 2]
```diff
@@ -136,101 +156,6 @@ def teardown_class(cls):
         cls.mailbox1.disconnect()
 
 
-# class TestFilteredSearchResult:
```
Comment (DavidMinarsch) [inline tests/test_connections/test_local/test_search_services.py]: We should uncomment this and add it back in.

File: tests/test_connections/test_local/test_search_services.py [hunk 3]
```diff
@@ -261,10 +188,49 @@ def test_unregister_service_result(self):
         result = OEFSerializer().decode(response_envelope.message)
         assert result.get("type") == OEFMessage.Type.OEF_ERROR
 
+        msg = OEFMessage(oef_type=OEFMessage.Type.REGISTER_SERVICE, id=0, service_description=service_description,
+                         service_id="Test_Service")
+        msg_bytes = OEFSerializer().encode(msg)
+        envelope = Envelope(to=DEFAULT_OEF, sender="mailbox1", protocol_id=OEFMessage.protocol_id, message=msg_bytes)
+        self.mailbox1.send(envelope)
+
+        """Test that at the beginning, the search request returns an empty search result."""
```
Comment (DavidMinarsch) [inline tests/test_connections/test_local/test_search_services.py]: Is this supposed to be a new test?

File: tests/test_connections/test_stub.py [hunk 4]
```diff
@@ -58,6 +60,20 @@ def test_reception(self):
...
         assert expected_envelope == actual_envelope
 
+    def test_connection_is_established(self):
+        """Test the stub connection is established and the bad formatted messages."""
+        assert self.connection.is_established
+        msg = DefaultMessage(type=DefaultMessage.Type.BYTES, content=b"hello")
+        encoded_envelope = "{},{},{},{}".format("any", "any", DefaultMessage.protocol_id, DefaultSerializer().encode(msg).decode("utf-8"))
+        encoded_envelope = base64.b64encode(encoded_envelope.encode("utf-8"))
+
+        self.connection._process_line(encoded_envelope)
```
Comment (DavidMinarsch) [inline tests/test_connections/test_stub.py]: Can you add another assert?

File: tests/test_connections/test_local/test_search_services.py (hunk 1 above)
Comment (Totoual) [reply tests/test_connections/test_local/test_search_services.py]: Yes ideally, I am registering an agent and then I have to search for this agent in order to complete the test. But we don't have the registering_agent yet. After we merge the register_agent - unregister_agent. I can complete the specific test.

File: tests/test_connections/test_local/test_search_services.py (hunk 2 above)
Comment (Totoual) [reply tests/test_connections/test_local/test_search_services.py]: Done.

**Pull request merged**